main.cpp

```diff
@@ -117,17 +143,58 @@ struct IntType
         delete value;
     }
 
-    operator int() { return *value; }
+    operator int() const { return *value; }
     
     IntType& add(int v);
     IntType& subtract(int v);
     IntType& multiply(int v);
     IntType& divide(int v);
+    
+    IntType& pow(const IntType& v);
+    IntType& pow(const FloatType& v);
+    IntType& pow(const DoubleType& v);
+    IntType& pow(int);
+    
+    IntType& powInternal(int v);
 
 private:
     int* value { nullptr };
 };
 
+// Point class
+Point::Point(float val_x, float val_y) : x(val_x), y(val_y) {}
+Point::Point(const DoubleType& val_x, const DoubleType& val_y) : x(static_cast<float>(val_x)), y(static_cast<float>(val_y)) {}
```

Comment:
demonstrate knowledge of forwarding constructors using your UDTs as arguments
i.e. 
```
Point::Point(int a) : Point(static_cast<float>(a), static_cast<float>(a)) { }
```
notice how `Point(int)` forwards its arguments to the 2-param `Point(float, float)` constructor 
- [ ] revise